src/array.cpp

```diff
@@ -81,5 +81,23 @@ namespace sparrow
     {
         return std::move(p_array);
     }
+
+    bool operator==(const array& lhs, const array& rhs)
+    {
+        return lhs.visit([&rhs](const auto& typed_lhs) -> bool
+        {
+            return rhs.visit([&typed_lhs](const auto& typed_rhs) -> bool
+            {
+                if constexpr (!std::same_as<decltype(typed_lhs), decltype(typed_rhs)>)
+                {
+                    return false;
+                }
+                else
+                {
+                    return typed_lhs == typed_rhs;
```

Comment:
I am a bit puzzled by this. Is == implemented by each layout?

Reply:
Layouts inheriting from `array_crtp_base` get it for free. The other layouts must implement it, I'll check if it's the case for each of them.

Reply:
Checked, all the layouts inherit or implement `operator==`.